tensorboard/components/vz_line_chart/vz-line-chart.ts

```diff
@@ -534,23 +571,36 @@ class LineChart {
   }
 
   private resetXDomain() {
-    // (Copied from DragZoomLayer.unzoom.)
-    const xScale = this.xScale as any;
-    xScale._domainMin = null;
-    xScale._domainMax = null;
-    const xDomain = xScale._getExtent();
+    let xDomain;
+    if (this._defaultXRange) {
```

Comment:
I think it would be slightly cleaner to not depend on implicit truthiness of arrays (i dislike reasoning about JS truthiness),
 i.e.: `this._defaultXRange != null`

Reply:
Done. The thing with explicit type-checking is that I keep forgetting whether to compare against null or undefined ... :)